react-native-plaid-link-sdk.d.ts

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import { TouchableOpacity, TouchableOpacityProps } from 'react-native';
+
+export interface Account {
+  id: string;
+  mask: string;
+  name: string;
+  subtype: string;
+  type: string;
+}
+
+export interface Institution {
+  institution_id: string;
+  name: string;
+}
+
+export interface OnSuccessMetadata {
+  status: string;
+  link_session_id: string;
+  request_id: string;
+  public_token: string;
+  account_id: string;
+  institution: Institution;
+  accounts: Array<Account & {
+    verification_status?: {
+      pending_automatic_verification: boolean;
+      pending_manual_verification: boolean;
+      manually_verified: boolean;
+    };
+  }>;
+  account: Account
+}
+
+export type ErrorStatus =
+  | 'requires_questions'
+  | 'requires_selections'
+  | 'requires_code'
+  | 'choose_device'
+  | 'requires_credentials'
+  | 'institution_not_found';
+
+export interface OnExitMetadata {
+  link_session_id: string;
+  request_id: string;
+  status: ErrorStatus;
+  institution: Institution;
+}
+
+export type EventName =
+  | 'ERROR'
+  | 'EXIT'
+  | 'HANDOFF'
+  | 'OPEN'
+  | 'OPEN_MY_PLAID'
+  | 'SEARCH_INSTITUTION'
+  | 'SELECT_INSTITUTION'
+  | 'SUBMIT_CREDENTIALS'
+  | 'SUBMIT_MFA'
+  | 'TRANSITION_VIEW';
+
+export type ViewName =
+  | 'CONNECTED'
+  | 'CONSENT'
+  | 'CREDENTIAL'
+  | 'ERROR'
+  | 'EXIT'
+  | 'LOADING'
+  | 'MFA'
+  | 'NUMBERS'
+  | 'RECAPTCHA'
+  | 'SELECT_ACCOUNT'
+  | 'SELECT_INSTITUTION';
+
+export interface EventMetadata {
+  error_code: string;
+  error_message: string;
+  error_type: string;
+  exit_status: string;
+  institution_id: string;
+  institution_name: string;
+  institution_search_query: string;
+  request_id: string;
+  link_session_id: string;
+  mfa_type: string;
+  view_name: ViewName;
+  timestamp: string;
+}
+
+export interface OnEventArgs {
+  eventName: EventName;
+  metadata: EventMetadata;
+}
+
+export type Product =
+  | 'auth'
+  | 'identity'
+  | 'income'
+  | 'transactions'
+  | 'assets'
+  | 'liabilities'
+  | 'investments';
+
+declare module 'react-native-plaid-link-sdk' {
+  export interface PlaidLinkProps<
+    C = TouchableOpacity,
+    P = TouchableOpacityProps
+  > {
```

Comment:
I've typed pieces of this library in the past for work, and you should be able to simplify this a bit so it doesn't need two generic arguments, and also allows enforcing that the type of the component props actually matches for the component. I ended up with something like:

```
export interface PlaidLinkProps<T> {
  ...otherProps,
  component?: (props: T) => React.ReactElement;
  componentProps?: T;
}

class PlaidLink<T> extends Component<PlaidLinkProps<T>> {}
export default PlaidLink;
```

Might even be able to push it a bit further and require that `T` extends something like `{ onPress: () => void }`, because otherwise it'll be unclickable (which is somewhat common sense, but not clear from the documentation here).